_posts/2016-10-10-Help!-I'm-becoming-Post-Junior.md

```diff
@@ -0,0 +1,162 @@
+
+> ▪	Makes steady, well-paced progress without the need for constant significant feedback from more senior engineers.
+
+> ▪	Owns a small-to-medium feature from technical design through completion.
+
+> ▪	Provides help and support outside of area under their responsibility.
+
+What can we gleam from this, that we start thinking of ways to improve ourselves as individual contributors?
+
+## Being Aware of How The Sausage Is Made
+
+It’s very unlikely that you are making all of the calls in a product, what is hopefully happening is that your team figures out a plan to ship something in a reasonable timeframe and as a team, you assign each other smaller talks that make that work. 
+
+It’s very likely that as a junior, you will be given the most well-defined small tasks. In OSS we call these the “[easy first steps][2].” tasks that can be done atomically, without requiring more interaction with designers or members of the product team. This is a feature, that eventually turns into a bug. 
+
+When it’s time to divvy up responsibilities, you should consider speaking up about taking tasks that are blocking people, but require further investigation outside of your dev team.
+
+This could be checking up occasionally with an API team, or leasing with different parts of the business to get confirmation on a specific detail. These kind of skills are not the sort of things you get in a computer science degree, and those without them are in a great place to other skills they’ve learned.
+
+## Shipping 
+
+When you work, you’re probably working inside an application that other people have built. They have laid the frameworks down, established the team norms and architectural choices that could have existed for years before you arrived. 
+
+Part of what gives you that feeling of confidence in your code is you’re comfortable within the architectural ecosystem you’re used to: e.g. React + Relay, iOS MVVM, Backbone + Express, Rails + CoffeeScript. You’re probably getting good at using them, and that’s awesome.
+
+To evolve from just that you need to really understand why these choices were made, what their trade-offs are and what was the reasoning for that platform to even exist in the first place.
+
+By understanding the history of the choices that you have been living with, you can make better decisions in the future.
+
+## Spit Shine
+
+There is no “One True Solution” for anything in programming. You should be wary of anyone that tries to tell you all similar sounding things should be done one way. The things you work on are likely a combination of different patterns that work together to become “an app.”
+
+Understanding your patterns well, and knowing when to apply them will get you far. If you want to go further still, you need to be able to step back from your patterns and try to see larger pictures. A feature that you write this week  _with a little bit more abstraction today_ could make it much easier to write something next week. 
```

Comment:
Great content. Maybe adding how it relates to the eng ladder mentioned in the beginning? e.g. adding "This is what will allow you to grow into becoming a multiplier for other people's work"